libshaderc_spvc/src/spvcir_pass.cc

```diff
@@ -127,6 +162,36 @@ void SpvcIrPass::GenerateSpirvCrossIR(Instruction *inst) {
       break;
     }
 
+    // opcode: 71
+    case SpvOpDecorate:
+    // opcode: 332
+    case SpvOpDecorateId: {
+      // spirv-cross comment:
+      // OpDecorateId technically supports an array of arguments, but our only
+      // supported decorations are single uint, so merge decorate and
+      // decorate-id here.
+      // TODO(sarahM0): investigate if this limitation is acceptable.
+      uint32_t id = inst->GetSingleWordInOperand(0u);
+
+      auto decoration =
+          static_cast<spv::Decoration>(inst->GetSingleWordInOperand(1u));
+      if (inst->NumInOperands() > 2) {
+        // instruction offset + length = offset_ + 1 + inst->NumOpreandWords()
+        assert(offset_ + 1 + inst->NumOperandWords() < ir_->spirv.size() &&
```

Comment:
Is this condition possible for a validly parsed shader or does it being violated indicate a bug in the parsing?

If it is possible for a valid shader to violate this assert, then there should be some sort of guard logic down below, since on non-debug builds the assert will be compiled out. If it is just a sanity check on the parser, then it is fine to leave as an assert

Reply:
It's a sanity check (having safe guards for valid shaders violations is very good point and I'm going to keep it mind).